hibernate4/src/main/java/com/fasterxml/jackson/datatype/hibernate4/HibernateProxySerializer.java

```diff
@@ -194,4 +200,40 @@ protected Object findProxied(HibernateProxy proxy)
         }
         return init.getImplementation();
     }
+    
+    // Alas, hibernate offers no public api to access this information, so we must resort to ugly hacks ...
+    protected static class ProxyReader {
+
+        // static final so the JVM can inline the lookup
+        private static final Field getIdentifierMethod;
+
+        static {
+            try {
+                getIdentifierMethod = BasicLazyInitializer.class.getDeclaredField("getIdentifierMethod");
+                getIdentifierMethod.setAccessible(true);
+            } catch (Exception e) {
+            	// should never happen: the field exists in all versions of hibernate 4 and 5
+                throw new RuntimeException(e); 
+            }
+        }
+
+        /**
+         * @return the name of the identifier property, or null if the name could not be determined
+         */
+        static String getIdentifierPropertyName(LazyInitializer init) {
+            try {
+                Method idGetter = (Method) getIdentifierMethod.get(init);
```

Comment:
LazyInitializer itself not generated (it is the Javassist MethodHandler the generated proxy class delegates to, which has special handling for the getter of the persistent identity).

I use reflection here because the field is private and has no getter :-(